src/lib/Repository/DecoratedLocationService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository;
+
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use EzSystems\HybridPlatformUi\Decorator\LocationDecorator;
+
+/**
+ * Service for loading locations with additional data not provided by the original API.
+ * Returns decorated location objects to facilitate storing the additional data without modifying the original object.
+ */
+class DecoratedLocationService
+{
+    /**
+     * @var LocationService
+     */
+    private $locationService;
+
+    /**
+     * @var PathService
+     */
+    private $pathService;
+
+    public function __construct(LocationService $locationService, PathService $pathService)
+    {
+        $this->locationService = $locationService;
+        $this->pathService = $pathService;
+    }
+
+    /**
+     * Loads all locations for $contentInfo with each location wrapped by a decorator.
+     * Additionally retrieves and sets child count, path locations and whether a location is main.
+     * Intended usage is admin only, where we display information that isn't provided by the standard location service.
+     *
+     * @param ContentInfo $contentInfo
+     *
+     * @return LocationDecorator[]
+     */
+    public function loadLocations(ContentInfo $contentInfo)
+    {
+        $locationService = $this->locationService;
+        $pathService = $this->pathService;
+        $locations = $locationService->loadLocations($contentInfo);
+
+        $decoratedLocations = array_map(
```

Comment:
Could you refactor this block to a private method, so that we know at a glance what it does ?